### PR TITLE
Store IRIS requests in our database

### PR DESCRIPTION
**Pull request by bitboxer**
closes https://github.com/railslove/recover-backlog/issues/129

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,6 +8,9 @@
   mount RailsAdmin::Engine => '/admin', as: 'rails_admin'
   mount Sidekiq::Web => '/sidekiq'
 
+  # TODO: make sure this route is only accessible from the EPS server, not the internet
```
Comment (bonflintstone) [inline config/routes.rb:11]: Where is the EPS server hosted at the moment?

File: app/controllers/iris_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'jimson'
+
+class IrisController
+  extend Jimson::Handler
+
+  def createDataRequest(params)
+    company = Company.find(params["dataRequest"]["locationID"])
+    return {"_" => "ERROR: Location not Found"} unless company
```
Comment (bonflintstone) [inline app/controllers/iris_controller.rb:8]: `find` would throw an exception, so this handling should probably go in a `rescue` clause

File: app/controllers/iris_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    data_request = DataRequest.new(
+      from: params["dataRequest"]["start"],
+      to: params["dataRequest"]["end"],
+      reason: params["dataRequest"]["requestDetails"],
+      iris_client_name: params["_client"]["name"],
+      iris_data_authorization_token: params["dataRequest"]["dataAuthorizationToken"],
+      iris_connection_authorization_token: params["dataRequest"]["connectionAuthorizationToken"],
+      company: company
+    )
+    data_request.save!
```
Comment (bonflintstone) [inline app/controllers/iris_controller.rb:19]: Might as well use `create!`

**Review by bonflintstone — APPROVED** [review]

File: config/routes.rb (hunk 1 above)
Comment (bitboxer) [reply config/routes.rb:11]: Nowhere. Someone(TM) has to build it :D. I will write down what the status is and what needs to be done in our slack channel in a couple of minutes.

Comment (salimhb) [reply config/routes.rb:11]: @bitboxer @hey-johnnypark do you think it's enough to protect this using HTTP Basic Authentication?

Comment (bitboxer) [reply config/routes.rb:11]: You cannot add basic auth because the eps server cannot talk that...at least I think it is not.

Comment (salimhb) [reply config/routes.rb:11]: I was thinking of simply adding them to the configured URL: `https://user:password@api.rcvr.app`

**Pull request merged**